module2/src/com/epam/decomposition/NumbersAnalyzer.java

```diff
@@ -0,0 +1,292 @@
+package com.epam.decomposition;
+
+
+import java.util.ArrayList;
+
+public class NumbersAnalyzer {
+    //task 1
+    public static int findLeastCommonMultiple(int firstNumber, int secondNumber) {
+        return ((firstNumber * secondNumber) / findGreatestCommonDivisor(firstNumber, secondNumber));
+    }
+
+    private static int findGreatestCommonDivisor(int firstNumber, int secondNumber) {
+        while (firstNumber != secondNumber) {
+            if (firstNumber > secondNumber) {
+                firstNumber = firstNumber - secondNumber;
+            } else {
+                secondNumber = secondNumber - firstNumber;
+            }
+        }
+        return firstNumber;
+    }
+
+    private static int findGreatestCommonDivisor(int firstNumber, int secondNumber, int thirdNumber, int fourthNumber) {
+        return findGreatestCommonDivisor((findGreatestCommonDivisor(firstNumber, secondNumber)),
+                (findGreatestCommonDivisor(thirdNumber, fourthNumber)));
+        }
+
+    //task 2
+    public static int findLeastCommonMultiple(int firstNumber, int secondNumber,
+                                                            int thirdNumber, int fourthNumber) {
+        return findGreatestCommonDivisor(firstNumber,secondNumber,thirdNumber,fourthNumber);
+    }
+
+    //task 5
+    public static int findSecondLargestNumber(int[] array) {
+        int max = array[0];
+        int secondMax = 0;
+            for (int i = 0; i < array.length; i++) {
+                if (array[i] > max) {
+                    secondMax = max;
+                    max = array[i];
+                }
```

Comment:
You also need to check if you numbers[i] > secondMax
```suggestion
                } else if (array[i] > secondMax) {
                secondMax = array[i];
            }
```